src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -228,30 +228,12 @@ internal static CompletionResult GetCommandNameCompletionResult(string name, obj
             }
 
             syntax = string.IsNullOrEmpty(syntax) ? name : syntax;
-            bool needAmpersand;
+            name = CodeGeneration.QuoteArgument(name, quote == string.Empty ? (char)0 : quote[0], false);
```

Comment:
Would it be acceptable to rewrite `HandleSingleAndDoubleQuote()`:

https://github.com/PowerShell/PowerShell/blob/8f5b2c241ec1bc0e9c97176b591cf342dcc10da2/src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs#L6489-L6519

with this:

```C#
        internal static char HandleDoubleAndSingleQuote(ref string wordToComplete)
        {
            char quote = (char)0;

            if (!string.IsNullOrEmpty(wordToComplete) && (wordToComplete[0].IsSingleQuote() || wordToComplete[0].IsDoubleQuote()))
            {
                quote = wordToComplete[0].IsSingleQuote() ? '\'' : '"';
                int length = wordToComplete.Length;

                wordToComplete = length == 1 ? string.Empty :
                    (quote.IsSingleQuote() ? wordToComplete[length - 1].IsSingleQuote() : wordToComplete[length - 1].IsDoubleQuote()) ?
                        wordToComplete.Substring(1, length - 2) : wordToComplete.Substring(1);

            }

            return quote;
        }
```
and then eliminate the kludge?  This would change the entire code over to using `char` to pass the quote.

Reply:
You check quote == 0 only once in "QuoteArgument". I guess you could use quote.Empty and don't use the zero value.

Reply:
It comes down to, do you use a string to store just three values, empty, `'` or `"`. or is there a better way?  Using a string means you have to use [0] references to get at the character (after you've confirmed that the string isn't empty) for using `IsSingleQuote` and `IsDoubleQuote`.

In a sense, its a glorified enum.  Is(Single/Double)Quote's are overkill because at this point it can only contain 3 values, not any of the other 7 quotes (at least as defined by HandleSingleAndDoubleQuote).